lib/rules/media-feature-range-notation/index.js

```diff
@@ -33,57 +40,101 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, ({ parent, value }) => {
-				if (!isStandardSyntaxMediaFeatureName(value)) return;
-
-				if (!isRangeContextMediaFeature(value) && isInBooleanContext(parent)) return;
-
-				if (!isRangeContextMediaFeature(value) && !isRangeTypeMediaFeature(value)) return;
-
-				if (primary === 'prefix' && isPrefixedRangeMediaFeature(value)) return;
+			const mediaQueryList = parseMediaQuery(atRule, result);
+			let hasFixes = false;
+
+			mediaQueryList.forEach((mediaQuery) => {
+				mediaQuery.walk((entry) => {
+					const node = entry.node;
+
+					// Only look at plain and range notation media features
+					if (!isMediaFeatureRange(node) && !isMediaFeaturePlain(node)) return;
+
+					// Expected plain notation and received plain notation
+					if (primary === 'prefix' && isMediaFeaturePlain(node)) return;
+
+					// Expected range notation and received range notation
+					if (primary === 'context' && isMediaFeatureRange(node)) return;
+
+					const featureName = node.getName();
+					const unprefixedMediaFeature = featureName.replace(/^(?:min|max)-/i, '');
+
+					if (!rangeTypeMediaFeatureNames.has(unprefixedMediaFeature)) return;
+
+					if (context.fix && primary === 'context' && isMediaFeaturePlain(node)) {
+						hasFixes = true;
+
+						/** @type {import('@csstools/css-tokenizer').TokenDelim} */
+						let operator = [TokenType.Delim, '<', -1, -1, { value: '<' }];
+
+						if (/^min-/i.test(featureName)) {
+							operator = [TokenType.Delim, '>', -1, -1, { value: '>' }];
+						}
+
+						const parent = entry.parent;
+
+						if (!isMediaFeature(parent)) return;
+
+						parent.feature = new MediaFeatureRangeNameValue(
+							new MediaFeatureName(
+								new TokenNode([
+									TokenType.Ident,
+									unprefixedMediaFeature,
+									-1,
+									-1,
+									{ value: unprefixedMediaFeature },
+								]),
+								node.name.before,
+								node.name.after.length > 0
+									? node.name.after
+									: [[TokenType.Whitespace, ' ', -1, -1, undefined]],
+							),
+							[operator, [TokenType.Delim, '=', -1, -1, { value: '=' }]],
+							node.value,
+						);
```

Comment:
Creating a new `MediaFeatureRangeNameValue` is a bit elaborate and painful at this moment, but having easy access to the low level API makes it trivial to have the right whitespace around and in between all the parts.

If I were to create a high level API in the upstream package it would follow the CSS spec serialization rules and it wouldn't work the same.

This might be necessary complexity.